### PR TITLE
:trollface:

### PR DESCRIPTION
**Pull request by AAGaming00**
Who needs `isDeveloper`.

### CODE REVIEW

**Comment by AAGaming00** [comment]
this doesnt allow setting overrides, need to fix it

**Comment by AAGaming00** [comment]
Updated.

**Comment by AsukeAZ** [comment]
this doesn't work

**Comment by Inve1951** [comment]
it may have
discord is playing cat&mouse with breaking experiments for us right now

**Comment by AsukeAZ** [comment]
yeah i don't know why they did that

**Comment by AsukeAZ** [comment]
i was using a lot of features with the plugin now i can't sadly

**Comment by AAGaming00** [comment]
> it may have discord is playing cat&mouse with breaking experiments for us right now

Checked the diffs, they haven't patched this method yet.

**Comment by AAGaming00** [comment]
Should be fixed.

**Comment by ReeceDonovan** [comment]
> Should be fixed.

image

**Comment by XxUnkn0wnxX** [comment]
Tested it & it only allows you to view Experiments, not interact/change any of the experiments.

**Comment by AsukeAZ** [comment]
thx u men

**Comment by XxUnkn0wnxX** [comment]
yep all good here ;)

**Comment by DanielGolan-mc** [comment]
when will this be merged? Can I use it before?

**Review by DanielGolan-mc — APPROVED** [review]
looks real to me.

**Comment by Inve1951** [comment]
I may merge it when discord stopped breaking it. No promises tho.

**Comment by XxUnkn0wnxX** [comment]
for `actionHandler["CONNECTION_OPEN"]` please use "OVERLAY_INITIALIZE" since "CONNECTION_OPEN" has issues when trying to upload files or images in chat it fails..

I've tested it works with "OVERLAY_INITIALIZE"
Credits to **ace19#1790** for suggesting it..

**Comment by Tharki-God** [comment]
experiment tab is not getting disabled when disabling the plugin

**Comment by D4rkov** [comment]
> OVERLAY_INITIALIZE

type shall be changed to `"OVERLAY_INITIALIZE"` as well?

**Comment by D4rkov** [comment]
The plugin stops working after some time, like you still have access to the experiments tab but you can't enable anything, and needs to be restarted.

**Comment by Orangenal-name** [comment]
the branch is 2 commits behind

**Comment by D4rkov** [comment]
@Inve1951 Just launched discord and I get this error.
image

File: plugins/discordexperiments.plugin.js [hunk 1]
```diff
@@ -1,28 +1,31 @@
...
 module.exports = class {
 	getName(){ return "Discord Experiments"; }
 
 	start() {
-    storeExports.default = new Proxy(original, {
-			get(_, key) {
-				return key === "isDeveloper" ? true : original[key];
-			}
-		});
+		const nodes = Object.values(settingsStore.default._dispatcher._dependencyGraph.nodes);
```
Comment (dav1312) [inline plugins/discordexperiments.plugin.js:18]:
```js
const nodes = Object.values(settingsStore.default._dispatcher._actionHandlers._dependencyGraph.nodes);
```

**Review by dav1312 — CHANGES_REQUESTED** [review]

**Comment by martijnmelchers** [comment]
Anyone know why Discord is trying to make it harder to access experiments?

**Comment by Tharki-God** [comment]
Because people are dumb

**Comment by Tharki-God** [comment]
okay , thanks for free promotion

**Pull request closed without merging**